plugins/inputs/cnosdb/kv_service.go

```diff
@@ -32,97 +31,96 @@ func NewTSKVService(acc telegraf.Accumulator) service.TSKVServiceServer {
+				case "TAG_STRING":
+					colArrMap[j] = array.NewStringData(col.Data())
+				case "FIELD_STRING":
+					colArrMap[j] = array.NewStringData(col.Data())
+				case "FIELD_BIGINT":
+					colArrMap[j] = array.NewInt64Data(col.Data())
+				case "FIELD_BIGINT UNSIGNED":
+					colArrMap[j] = array.NewUint64Data(col.Data())
+				case "FIELD_DOUBLE":
+					colArrMap[j] = array.NewFloat64Data(col.Data())
+				case "FIELD_BOOLEAN":
+					colArrMap[j] = array.NewBooleanData(col.Data())
 				}
+			}
 
+			for i := 0; i < int(numRows); i++ {
 				tags := make(map[string]string)
 				fields := make(map[string]interface{})
 
-				tagsBitSet := BitSet{
-					buf: point.TagsNullbitBytes(),
-					len: schema.TagNameLength(),
-				}
-				if point.TagsLength() > 0 {
-					for tagI := 0; tagI < point.TagsLength(); tagI++ {
-						if !tagsBitSet.get(tagI) {
-							continue
-						}
-						if !point.Tags(&tag, tagI) {
-							continue
-						}
-						tags[string(schema.TagName(tagI))] = string(tag.ValueBytes())
-					}
-				}
-				fieldsBitSet := BitSet{
-					buf: point.FieldsNullbitBytes(),
-					len: schema.FieldNameLength(),
-				}
-				if point.FieldsLength() > 0 {
-					for fieldI := 0; fieldI < point.FieldsLength(); fieldI++ {
-						if !fieldsBitSet.get(fieldI) {
-							continue
-						}
-						if !point.Fields(&field, fieldI) {
-							continue
-						}
-						switch schema.FieldType(fieldI) {
-						case models.FieldTypeInteger:
-							v := binary.BigEndian.Uint64(field.ValueBytes())
-							fields[string(schema.FieldName(fieldI))] = int64(v)
-						case models.FieldTypeUnsigned:
-							v := binary.BigEndian.Uint64(field.ValueBytes())
-							fields[string(schema.FieldName(fieldI))] = v
-						case models.FieldTypeFloat:
-							tmp := binary.BigEndian.Uint64(field.ValueBytes())
-							v := math.Float64frombits(tmp)
-							fields[string(schema.FieldName(fieldI))] = v
-						case models.FieldTypeBoolean:
-							v := field.ValueBytes()
-							fields[string(schema.FieldName(fieldI))] = v[0] == byte(1)
-						case models.FieldTypeString:
-							v := string(field.ValueBytes())
-							fields[string(schema.FieldName(fieldI))] = v
-						default:
-							// Do nothing ?
-						}
+				for j := 0; j < int(numCols); j++ {
+					switch columnIndexToType[uint64(j)] {
+					case "TAG_STRING":
+						tags[columnIndexToName[j]] = colArrMap[j].(*array.String).Value(i)
+					case "FIELD_STRING":
+						tags[columnIndexToName[j]] = colArrMap[j].(*array.String).Value(i)
+					case "FIELD_BIGINT":
+						fields[columnIndexToName[j]] = colArrMap[j].(*array.Int64).Value(i)
+					case "FIELD_BIGINT UNSIGNED":
+						fields[columnIndexToName[j]] = colArrMap[j].(*array.Uint64).Value(i)
+					case "FIELD_DOUBLE":
+						fields[columnIndexToName[j]] = colArrMap[j].(*array.Float64).Value(i)
+					case "FIELD_BOOLEAN":
+						fields[columnIndexToName[j]] = colArrMap[j].(*array.Boolean).Value(i)
 					}
 				}
 				s.accumulator.AddFields(tab, fields, tags)
```

Comment:
Maybe you can use accumulator.AddMetric to send timestamps extracted from the source:
```go
if hasTimestamp {
  s.accumulator.AddMetric(metric.New(tab, tags, fields, time))
} else {
  s.accumulator.AddFields(tab, fields, tags)
}
```